gobblin-restli/gobblin-flow-config-service/gobblin-flow-config-service-client/src/main/java/org/apache/gobblin/service/FlowExecutionClient.java

```diff
@@ -0,0 +1,170 @@
+
+  private Optional<HttpClientFactory> _httpClientFactory;
+  private Optional<RestClient> _restClient;
+  private final FlowexecutionsRequestBuilders _flowexecutionsRequestBuilders;
+
+  /**
+   * Construct a {@link FlowExecutionClient} to communicate with http flow execution server at URI serverUri
+   * @param serverUri address and port of the REST server
+   */
+  public FlowExecutionClient(String serverUri) {
+    LOG.debug("FlowConfigClient with serverUri " + serverUri);
+
+    _httpClientFactory = Optional.of(new HttpClientFactory());
+    Client r2Client = new TransportClientAdapter(_httpClientFactory.get().getClient(Collections.<String, String>emptyMap()));
+    _restClient = Optional.of(new RestClient(r2Client, serverUri));
+
+    _flowexecutionsRequestBuilders = createRequestBuilders();
+  }
+
+  /**
+   * Construct a {@link FlowExecutionClient} to communicate with http flow execution server at URI serverUri
+   * @param restClient restClient to send restli request
+   */
+  public FlowExecutionClient(RestClient restClient) {
+    LOG.debug("FlowConfigClient with restClient " + restClient);
+
+    _httpClientFactory = Optional.absent();
+    _restClient = Optional.of(restClient);
+
+    _flowexecutionsRequestBuilders = createRequestBuilders();
+  }
+
+  protected FlowexecutionsRequestBuilders createRequestBuilders() {
+    return new FlowexecutionsRequestBuilders();
+  }
+
+  /**
+   * Get a flow execution
+   * @param flowStatusId identifier of flow execution to get
+   * @return a {@link FlowExecution} with the flow execution
+   * @throws RemoteInvocationException
+   */
+  public FlowExecution getFlowExecution(FlowStatusId flowStatusId)
+      throws RemoteInvocationException {
+    LOG.debug("getFlowConfig with groupName " + flowStatusId.getFlowGroup() + " flowName " +
+        flowStatusId.getFlowName());
+
+    GetRequest<FlowExecution> getRequest = _flowexecutionsRequestBuilders.get()
+        .id(new ComplexResourceKey<>(flowStatusId, new EmptyRecord())).build();
+
+    Response<FlowExecution> response =
+        _restClient.get().sendRequest(getRequest).getResponse();
+    return response.getEntity();
+  }
+
+  /**
+   * Get the latest flow execution
+   * @param flowId identifier of flow execution to get
+   * @return a {@link FlowExecution}
+   * @throws RemoteInvocationException
+   */
+  public FlowExecution getLatestFlowExecution(FlowId flowId)
+      throws RemoteInvocationException {
+    LOG.debug("getFlowConfig with groupName " + flowId.getFlowGroup() + " flowName " +
+        flowId.getFlowName());
+
+    FindRequest<FlowExecution> findRequest = _flowexecutionsRequestBuilders.findByLatestFlowExecution().flowIdParam(flowId).build();
+
+    Response<CollectionResponse<FlowExecution>> response =
+        _restClient.get().sendRequest(findRequest).getResponse();
+
+    List<FlowExecution> flowExecutionList = response.getEntity().getElements();
```

Comment:
I don't think it's necessary since if there's no flow executions, it already will throw a wrapped rest exception when sending the request in the line before.